packages/components/src/components/pin-code/pin-code.tsx

```diff
@@ -0,0 +1,218 @@
+import { Component, Element, Event, type EventEmitter, h, Host, type JSX, Prop } from '@stencil/core';
+import type { BreakpointCustomizable, PropTypes, Theme, ValidatorFunction } from '../../types';
+import type { PinCodeState, PinCodeType, PinCodeUpdateEvent } from './pin-code-utils';
+import {
+  AllowedTypes,
+  attachComponentCss,
+  FORM_STATES,
+  hasDescription,
+  hasLabel,
+  hasMessage,
+  hasPropValueChanged,
+  isWithinForm,
+  THEMES,
+  validateProps,
+} from '../../utils';
+import { getComponentCss } from './pin-code-styles';
+import {
+  initHiddenInput,
+  inputIsSingleDigit,
+  joinInputValues,
+  PIN_CODE_TYPES,
+  syncHiddenInput,
+} from './pin-code-utils';
+import { StateMessage } from '../common/state-message/state-message';
+import { Required } from '../common/required/required';
+
+const propTypes: PropTypes<typeof PinCode> = {
+  label: AllowedTypes.string,
+  description: AllowedTypes.string,
+  length: AllowedTypes.number,
+  hideLabel: AllowedTypes.breakpoint('boolean'),
+  state: AllowedTypes.oneOf<PinCodeState>(FORM_STATES),
+  disabled: AllowedTypes.boolean,
+  required: AllowedTypes.boolean,
+  message: AllowedTypes.string,
+  theme: AllowedTypes.oneOf<Theme>(THEMES),
```

Comment:
Let's stick to the same order as props are defined down below instead of shuffling it around.